### ISSUE TITLE
[Improve Best Practice Guide]: Unit testing, et. al.

### ISSUE THREAD

**Issue by riverma** [NASA-AMMOS/slim#24]
### Checked for duplicates

Yes - I've already checked

### Best Practice Guide

Continuous Testing

### Best Practice Guide Sections

- [ ] Use Cases
- [ ] Trade Studies
- [X] Reference Architectures
- [X] Starter Kits

### Describe the improvement

We'd like to have starter kits for helping the SLIM community with getting a JUnit or PyUnit testing framework up and running, by providing real world example templates to customize and start using. Additionally, having an architectural reference diagram explaining how Java unit test using JUnit in Jenkins would ideally work would be great. These two were mentioned in [previous changelog reports](https://github.com/NASA-AMMOS/slim/commit/224a36e9d58d43b4881c202fa615e1ffb845f197#diff-06572a96a58dc510037d5efa622f9bec8519bc1beab13c9f251e97e657a9d4edL21-L26), but were pushed to be done later.

**Comment by riverma** [comment]
Maybe we need another ticket for these, but it'd be good to have recommendations / starter kits / tool pointers for:
- Unit testing
- Integration
- System
- Regression

**Comment by jpl-jengelke** [comment]
I will focus on covering those in the Reference Architecture first. Maybe we should split this one off from the Starter Kit?

**Comment by riverma** [comment]
Yeah - agreed @jpl-jengelke. Reference architecture would help lay the context, but separate starter kits (if it makes sense) would be a good approach.

**Comment by riverma** [comment]
+1'd by @ramesh-maddegoda (Unity, PDS)

**Comment by riverma** [comment]
@yunks128 - do you have some material within your #110 ticket that could help with this ticket?

**Comment by yunks128** [comment]
@riverma This ticket is precisely what I'm currently working on with #110, including reference architectures and starter kits!

**Comment by jpl-jengelke** [comment]
I have the JUnit tests complete.

**Comment by riverma** [comment]
This is being worked on within #110 by @yunks128 and @riverma.

**Comment by riverma** [comment]
Done in https://github.com/NASA-AMMOS/slim/issues/110
